src/borg/helpers/time.py

```diff
@@ -109,6 +109,48 @@ def format_timedelta(td):
     return txt
 
 
+def calculate_relative_offset(format_string, from_date, earlier=False):
+    """ Calculates offset based on a relative marker. 7d (7 days), 8m (8 months)
+        earlier: whether offset should be calculated to an earlier time.
+    """
+    if from_date is None:
+        from_date = archive_ts_now().date()
+
+    if format_string is None:
+        return from_date
+
+    day_offset_regex = re.compile(r'\d+d')
+    month_offset_regex = re.compile(r'\d+m')
+
+    day_offset_str = day_offset_regex.search(format_string)
+    if day_offset_str is not None:
+        day_offset = int(day_offset_str.group()[:-1])
+        day_offset *= -1 if earlier else 1
+        return from_date + timedelta(days=day_offset)
```

Comment:
the return value of `.search` is a match object, so using `..._str` is rather misleading here.

Just use `match` and `if match:`?

You could also have the days part of the regex as a named group, then you would not need the `[:-1]` hack, but could directly access it.